pyaedt/misc/toolkit_manager.py

```diff
@@ -0,0 +1,283 @@
+            installation_option_action.get() == "Offline" and selected_toolkit == "Custom"
+        ):
+            toolkit_name.config(state="normal")
+            input_file_label.config(text="Enter script path:")
+            input_file.config(state="normal")
+            installation_option_action.set("Offline")
+        else:
+            toolkit_name.config(state="disabled")
+            input_file_label.config(text="Enter wheelhouse path:")
+            input_file.config(state="normal")
+
+    toolkits_combo.bind("<<ComboboxSelected>>", update_page)
+
+    update_page()
+
+    return install_button, uninstall_button, input_file, toolkits_combo, toolkit_name
+
+
+def is_toolkit_installed(toolkit_name):
+    if toolkit_name == "Custom":
+        return False
+    script_file = os.path.normpath(os.path.join(package_dir, available_toolkits[toolkit_name]["toolkit_script"]))
+    return True if os.path.isfile(script_file) else False
+
+
+def open_window(window, window_name, open_source_toolkits):
+    """Open window action"""
+    if not hasattr(window, "opened"):
+        window.opened = True
+        window.title(window_name)
+        install_button, uninstall_button, input_file, toolkits_combo, toolkit_name = create_toolkit_page(
+            window, open_source_toolkits
+        )
+        root.minsize(500, 250)
+        return install_button, uninstall_button, input_file, toolkits_combo, toolkit_name
+    else:
+        window.deiconify()
+
+
+def __get_command_function(
+    is_install, toolkit_level, input_file, toolkits_combo, toolkit_name, install_button, uninstall_button
+):
+    return lambda: button_is_clicked(
+        is_install, toolkit_level, input_file, toolkits_combo, toolkit_name, install_button, uninstall_button
+    )
+
+
+def toolkit_window(toolkit_level="Project"):
+    """Main window"""
+    toolkit_window_var = tk.Toplevel(root)
+    open_source_toolkits = []
+    for toolkit_name, toolkit_info in available_toolkits.items():
+        if toolkit_info["installation_path"].lower() == toolkit_level.lower():
+            open_source_toolkits.append(toolkit_name)
+    toolkit_window_var.minsize(250, 150)
+    install_button, uninstall_button, input_file, toolkits_combo, toolkit_name = open_window(
+        toolkit_window_var, toolkit_level, open_source_toolkits
+    )
+
+    install_command = __get_command_function(
+        True, toolkit_level, input_file, toolkits_combo, toolkit_name, install_button, uninstall_button
+    )
+    uninstall_command = __get_command_function(
+        False, toolkit_level, input_file, toolkits_combo, toolkit_name, install_button, uninstall_button
+    )
+
+    install_button.configure(command=install_command)
+    uninstall_button.configure(command=uninstall_command)
+
+
+def button_is_clicked(
+    install_action, toolkit_level, input_file, combo_toolkits, toolkit_name, install_button, uninstall_button
+):
+    """Install/Uninstall button action"""
```

Comment:
Can descriptions be redone following our style? Perhaps something like this: "Set up a button for installing and uninstalling the toolkit."